FirebaseInstallations/Source/Tests/Unit/FIRInstallationsIDControllerTests.m

```diff
@@ -87,6 +87,46 @@ - (void)tearDown {
   self.appName = nil;
 }
 
+#pragma mark - Initialization
+
+- (void)testInitWhenProjectIDSetThenItIsPassedToAPIService {
+  NSString *APIKey = @"api-key";
+  NSString *projectID = @"project-id";
+  OCMExpect([self.mockAPIService alloc]).andReturn(self.mockAPIService);
+  OCMExpect([self.mockAPIService initWithAPIKey:APIKey projectID:projectID])
+      .andReturn(self.mockAPIService);
+
+  FIRInstallationsIDController *controller =
+      [[FIRInstallationsIDController alloc] initWithGoogleAppID:@"app-id"
+                                                        appName:@"app-name"
+                                                         APIKey:APIKey
+                                                      projectID:projectID
+                                                    GCMSenderID:@"sender-id"
+                                                    accessGroup:nil];
+  XCTAssertNotNil(controller);
+
+  OCMVerifyAll(self.mockAPIService);
+}
+
+- (void)testInitWhenProjectIDIsNilThenGCMSenderIDIsPassedToAPIServiceAsProjectID {
+  NSString *APIKey = @"api-key";
+  NSString *GCMSenderID = @"sender-id";
+  OCMExpect([self.mockAPIService alloc]).andReturn(self.mockAPIService);
+  OCMExpect([self.mockAPIService initWithAPIKey:APIKey projectID:GCMSenderID])
+      .andReturn(self.mockAPIService);
+
+  FIRInstallationsIDController *controller =
+      [[FIRInstallationsIDController alloc] initWithGoogleAppID:@"app-id"
+                                                        appName:@"app-name"
+                                                         APIKey:APIKey
+                                                      projectID:@""
+                                                    GCMSenderID:GCMSenderID
+                                                    accessGroup:nil];
+  XCTAssertNotNil(controller);
+
+  OCMVerifyAll(self.mockAPIService);
+}
```

Comment:
I didn't find a test that fails when no GCMSenderID and no projectID are set...
And a test that fails when API-Key is missing.

Reply:
They are [here](https://github.com/firebase/firebase-ios-sdk/pull/4683/files/5a74f2e0b02315a51593f374f1e21f15fc77d687#diff-81c148814f62ebc1039b03a2c28027e2R240)